packages/theming/apple-theme/src/createAppleTheme.macos.ts

```diff
@@ -2,15 +2,31 @@ import { ThemeReference } from '@fluentui-react-native/theme';
 import { Theme } from '@fluentui-react-native/theme-types';
 import { Appearance } from 'react-native';
 import { getBaseAppleThemeMacOS } from './appleTheme.macos';
+import { AccessibilityInfo } from 'react-native-macos';
+import { setIsHighContrast } from './appleHighContrast.macos';
+
+let appleThemeReference: ThemeReference;
 
 export function createAppleTheme(): ThemeReference {
-  const appleThemeReference = new ThemeReference({} as Theme, () => {
+  appleThemeReference = new ThemeReference({} as Theme, () => {
     return getBaseAppleThemeMacOS();
   });
+  // Fetch initial system settings for high contrast mode
+  highContrastHandler();
+  // Invalidate theme and set prop when high contrast setting changes
+  AccessibilityInfo.addEventListener('highContrastChanged', () => {
+    highContrastHandler();
+  });
 
   Appearance.addChangeListener(() => {
     appleThemeReference.invalidate();
   });
-
   return appleThemeReference;
 }
+
+function highContrastHandler() {
+  AccessibilityInfo.isHighContrastEnabled().then((isEnabled) => {
+    setIsHighContrast(isEnabled);
+    appleThemeReference.invalidate();
+  });
+}
```

Comment:
I wonder if we could encapsulate this logic ( + the added logic in createAppleTheme) into a hook like `useHighContrastEnabled()` or something, similar to this hook I found: https://github.com/react-native-hooks/accessibility-info/blob/master/src/index.js

Reply:
Not an expert on hooks but that seems like a cleaner approach. Though, aren't hooks meant to be reusable?

Reply:
This is also not in a component, which is what hooks are for.

Reply:
Ah yeah, the hook I linked relies on `useEffect` so it probably would not work here. Whelp.